_projects/geotrellis-vt-server.md

```diff
@@ -0,0 +1,57 @@
+---
+# Remove the project attributes which don't apply to your project. (ie: if no issues, delete the issues section)
+
+published: false # change to true once ready to post
+
+title: "Dynamic Vector Tile Server" # project title inside quotes
+excerpt: "Serve dynamic vector tiles from GeoJSON and other sources using GeoTrellis" # shows on project list page
+
+# project attributes
+
+requirements: # bullet list of requirements – one requirement per line – follow below format
+  - "experience building or using APIs"
+  - "interest in Scala / FP"
+
+tags: # one tag per line – spaces are allowed in tags – use tags other posts use
+  - "vector tiles"
+  - "scala"
+  - "GeoTrellis"
+
+difficulty: "medium" # Easy, Medium, Hard – pick one
+
+mentors: # github username without the @ – example: designmatty
+  - "jisantuc"
+  - "cloudniner"
+  
+---
+
+# Motivation
+
+Prior art exists showing efficient vector tile slicing and rendering for static data. Raster Foundry and other projects have a need for the ability to serve dynamically generated vector tiles from data provided by users as they use the application. Several other options (t-rex, tegola, tilegarden) require configuration to serve a set of layers when the server starts, but the goal of this project is for clients to ad hoc specify layers. This fellowship has the potential for open source contributions to two projects: [GeoTrellis](https://geotrellis.io/) and [Raster Foundry](https://www.rasterfoundry.com/).
+
+# Milestones
+
+## Preparation
+
+You will prepare by gaining a working knowledge of:
+
+  - scala development
+  - scala libraries (e.g. [http4s](https://http4s.org/), [doobie](https://tpolecat.github.io/doobie/), [circe](https://circe.github.io/circe/)) for web development
+  - GeoTrellis geometry libraries
+
+## Coding Phase 1 - Create Vector Tiles from a PostGIS Source
+
+In this phase, you will create functionality to serve vector tiles from vector data stored in a PostGIS database. 
```

Comment:
Is the plan for this work to be in backsplash, geotrellis-server, or geotrellis? Not sure if it matters for the proposal, just curious myself a little bit.

Reply:
I'm not sure -- if it's in backsplash, then it's a lot easier to test, since we have all these annotations and project extents lying around. But also if it's in backsplash then the development story is more complicated. On the other hand, gt-server is a more isolated development environment, but I don't think vector tiles fit super well with the arbitrary maml expression -> tile pattern

Reply:
I'd suggest that phase 1 and 2 would be its own project to keep things light and independent. In phase 3 there will be time to think about where to contribute the performance improvements and new abstractions (if any). I can easily see some code making it into GeoTrellis or geotrellis-contrib. Hopefully there will be room for integration into backsplash if results are encouraging, that could be a stretch goal?

Reply:
Having this project be its own project/repo makes sense to me and keeps the requirements low I agree. You are correct that if it looks promising, integration into the appropriate repositories could be a next phase or something we take on after the fellowship is complete.